contracts/HubPool.sol

```diff
@@ -231,6 +248,33 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
      *                ADMIN FUNCTIONS                *
      *************************************************/
 
+    /**
+     * @notice Pauses the bundle proposal and execution process. This is intended to be used during upgrades or when
+     * something goes awry.
+     * @param pause true if the call is meant to pause the system, false if the call is meant to unpause it.
+     */
+    function setPaused(bool pause) public onlyOwner nonReentrant {
+        paused = pause;
+        emit Paused(pause);
+    }
+
+    /**
+     * @notice This allows for the deletion of the active proposal in case of emergency.
+     * @dev This is primarily intended to rectify situations where an unexecutable bundle gets through liveness in the
+     * case of a non-malicious bug in the proposal/dispute code. Without this function, the contract would be
+     * indefinitely blocked, migration would be required, and in-progress transfers would never be repaid.
+     */
+    function emergencyDeleteProposal() public onlyOwner nonReentrant {
+        if (!rootBundleProposal.proposerBondRepaid) bondToken.safeTransfer(rootBundleProposal.proposer, bondAmount);
+        emit EmergencyRootBundleDeleted(
+            rootBundleProposal.poolRebalanceRoot,
+            rootBundleProposal.relayerRefundRoot,
+            rootBundleProposal.slowRelayRoot,
+            rootBundleProposal.proposer
+        );
+        delete rootBundleProposal;
```

Comment:
should we be `delete`-ing the proposal first before making the external `safeTransfer` call? I'm thinking of the check-effects-interaction pattern

Reply:
good point! transfer last.

Reply:
Hm i'd still rather this folows check-effects-interaction since there's no downside to doing so.

Reply:
Deleting first just means we have to store some temp variables, making the implementation a bit more messy. Happy to do it if we think it's important, though.

Reply:
Ah ok no problem

Reply:
As long as this is reentrancy guarded